src/components/ChangePlan/PlanCalculator/BannerUpgrade/BannerUpgrade.js

```diff
@@ -12,16 +12,28 @@ export const BannerUpgrade = ({ sessionPlan, currentPlan, currentPlanList }) =>
     switch (type) {
       case 'prepaid':
       case 'subscribers':
-        return {
-          banner: {
-            ...bannerInfo,
-            link: `/plan-selection/${currentPlan.featureSet}/monthly-deliveries`,
-          },
-          suggestion: {
-            ...suggestionInfo,
+        if (sessionPlan.planSubscription > 1) {
```

Comment:
I understand, but in that case I have to modify the component properties, and the purpouse will be the same. We already have **_planSubscription_** property which has the periodicity of the current plan. In my opinion, i wouldn't change it.

Reply:
It would be nice having this condition localized in only one section. Imagine that the administration issue with the subscriptions is resolved, then we will have to track all this ifs and change them. If we could have this behaviour only in the service, wouldn't it be great?